app/runtime/runtime/src/main/java/io/syndesis/integration/runtime/LoggingInterceptStrategy.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.integration.runtime;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import io.syndesis.core.KeyGenerator;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.model.LogDefinition;
+import org.apache.camel.model.ProcessorDefinition;
+import org.apache.camel.spi.InterceptStrategy;
+import org.apache.camel.spi.LogListener;
+import org.apache.camel.spi.UuidGenerator;
+import org.apache.camel.util.CamelLogger;
+
+import java.io.StringWriter;
+
+/**
+ * Created by chirino on 1/10/18.
+ */
+public class LoggingInterceptStrategy implements InterceptStrategy {
+
+
+    public UuidGenerator getUuidGenerator() {
+        return new UuidGenerator() {
+            @Override
+            public String generateUuid() {
+                // Generates always incrementing lexically sortable unique keys
+                return KeyGenerator.createKey();
+            }
+        };
+    }
+
+    public static class LogLine {
+        public final String id;
+        public final String tx;
+        public final String step;
+        public final String action;
+        public final String message;
+
+        public LogLine(String tx, String step, String action, String message) {
```

Comment:
Shouldn't a log line alos contain a date ?

Reply:
generated keys have an embedded timestamp in them.   See:
https://github.com/chirino/syndesis/blob/52137b104e872ee3ee30a116b703362fb68b7188/app/core/core/src/main/java/io/syndesis/core/KeyGenerator.java#L88